apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -43,19 +43,44 @@ function ModuleViewWrapper({
     return <ModuleView id={id} name={name} code={code} />;
 }
 
+const getModule = (
+    searchParams: URLSearchParams,
+    setSearchParams: (params: URLSearchParams) => void,
+    modulenames: string[]
+): string => {
+    const paramModule = searchParams.get('module') || null;
+
+    // A module has been specified and the package has this module
+    if (!!paramModule && modulenames.includes(paramModule)) {
+        return paramModule;
+    }
+
+    // A module has been specified and the package does not have this module
+    if (!!paramModule) {
+        const newSearchParams = new URLSearchParams(searchParams);
+        newSearchParams.delete('module');
+        setSearchParams(newSearchParams);
```

Comment:
I wouldn’t set during render at all, if we want to do this it can be in an effect.